File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        a = {}
        b = {}

        for i in range(len(s)):
            s_char = s[i]

            if s_char in a:
                a[s_char] += 1
            else:
                a[s_char] = 1
            
        for i in range(len(t)):
            t_char = t[i]
            
            if t_char in b:
                b[t_char] += 1
            else:
                b[t_char] = 1

        return a == b

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        
        vector<int> first(26, 0), second(26, 0);

        for (int i = 0; i < s.size(); i++){
            first[s[i] - 'a']++;
            second[t[i] - 'a']++;
        }

        return (first == second);
    }
};